packages/vite/src/node/plugins/resolve.ts

```diff
@@ -77,168 +77,177 @@ export function resolvePlugin(baseOptions: InternalResolveOptions): Plugin {
   }
   let server: ViteDevServer | undefined
 
-  return {
-    name: 'vite:resolve',
+  const resolveId: Plugin['resolveId'] = (id, importer, resolveOpts, ssr) => {
+    const targetWeb = !ssr || ssrTarget === 'webworker'
+
+    // this is passed by @rollup/plugin-commonjs
+    const isRequire =
+      resolveOpts &&
+      resolveOpts.custom &&
+      resolveOpts.custom['node-resolve'] &&
+      resolveOpts.custom['node-resolve'].isRequire
```

Comment:
Why such a long statement?

```suggestion
    const isRequire = resolveOpts?.custom?.['node-resolve']?.isRequire
```

Reply:
Not my changes :P